sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -178,8 +179,17 @@ private[parquet] class ParquetRowConverter(
 
   // Converters for each field.
   private[this] val fieldConverters: Array[Converter with HasParentContainerUpdater] = {
+    val catalystFieldNameToIndex = if (caseSensitive) {
+      catalystType.fieldNames.zipWithIndex.toMap
+    } else {
+      CaseInsensitiveMap(catalystType.fieldNames.zipWithIndex.toMap)
+    }
     parquetType.getFields.asScala.map { parquetField =>
-      val fieldIndex = catalystType.fieldIndex(parquetField.getName)
+      val fieldIndex = catalystFieldNameToIndex.getOrElse(parquetField.getName,
+        throw new RuntimeException(
```

Comment:
how can this happen?

Reply:
AFAIK for top-level columns, if users specify a non-existing column, the reader fills null for it. Are nested fields different?

Reply:
I'll remove it and apply its key directly. Tested by now using nested structure, it fills null properly.